core/storage/file/gae_models_test.py

```diff
@@ -0,0 +1,179 @@
+
+    def test_commit(self):
+        file_metadata_model = file_models.FileMetadataModel.create(
+            'exp_id1', 'path/to/file1.png')
+        self.assertEqual(file_metadata_model.version, 0)
+
+        file_metadata_model.commit(feconf.SYSTEM_COMMITTER_ID, [])
+        self.assertEqual(file_metadata_model.version, 1)
+
+
+class FileModelTest(test_utils.GenericTestBase):
+    """Tests the FileModel class."""
+
+    def test_get_new_id(self):
+        with self.assertRaises(NotImplementedError):
+            file_models.FileModel.get_new_id('entity1')
+
+    def test_construct_id(self):
+        file_model_id1 = file_models.FileModel._construct_id( # pylint: disable=protected-access
+            'exp_id1', 'path/to/file.png')
+        file_model_id2 = file_models.FileModel._construct_id( # pylint: disable=protected-access
+            'exp_id1', '/exp_id1/path/to/file.png')
+
+        self.assertEqual(file_model_id1, '/exp_id1/path/to/file.png')
+        self.assertEqual(file_model_id2, '/exp_id1/path/to/file.png')
+
+    def test_create(self):
+        file_model = file_models.FileModel.create(
+            'exp_id1', 'path/to/file.png')
+        self.assertEqual(file_model.deleted, False)
+        self.assertEqual(file_model.id, '/exp_id1/path/to/file.png')
+
+    def test_get_model(self):
+        file_model = file_models.FileModel.create(
+            'exp_id1', 'path/to/file1.png')
+        file_model.commit(feconf.SYSTEM_COMMITTER_ID, [])
+        actual_model = file_models.FileModel.get_model(
+            'exp_id1', 'path/to/file1.png')
+
+        self.assertEqual(actual_model, file_model)
+
+        actual_model = file_models.FileModel.get_model(
+            'exp_id1', 'path/to/file2.png')
+        self.assertEqual(actual_model, None)
+
+        with self.assertRaisesRegexp(Exception, (
+            'Entity for class FileModel '
+            'with id /exp_id1/path/to/file2.png not found')):
+            actual_model = file_models.FileModel.get_model(
+                'exp_id1', 'path/to/file2.png', True)
+
+    def test_get_version(self):
+        file_model = file_models.FileModel.create(
+            'exp_id1', 'path/to/file1.png')
+        file_model.commit(feconf.SYSTEM_COMMITTER_ID, [])
+
+        actual_model = file_models.FileModel.get_version(
+            'exp_id1', 'path/to/file1.png', 1)
+        self.assertEqual(file_model.key, actual_model.key)
+
+        file_model.commit(feconf.SYSTEM_COMMITTER_ID, [])
+        actual_model = file_models.FileModel.get_version(
+            'exp_id1', 'path/to/file1.png', 2)
+        self.assertEqual(file_model.key, actual_model.key)
+
+    def test_commit(self):
+        file_model = file_models.FileModel.create(
+            'exp_id1', 'path/to/file1.png')
+        self.assertEqual(file_model.version, 0)
+
+        file_model.commit(feconf.SYSTEM_COMMITTER_ID, [])
+        self.assertEqual(file_model.version, 1)
```

Comment:
Line break at the end of file.

Reply:
My bad! (Github doesn't shows line breaks at the end).

@ankita240796 I feel you missed tests for some functions and also could you please write the tests in the order as the functions are written in `gae_modals.py`. (Like the test_commit and test_get_version are swapped!)

Reply:
Done.